src/Integration.Vsix/Delegates/SonarAnalyzerStandaloneWorkflow.cs

```diff
@@ -0,0 +1,95 @@
+{
+    internal class SonarAnalyzerStandaloneWorkflow : SonarAnalyzerWorkflowBase
+    {
+        private readonly Func<IAnalysisRunContext, bool> previousShouldExecuteRuleFunc;
+        private readonly Action<IReportingContext> previousReportDiagnosticAction;
+
+        public SonarAnalyzerStandaloneWorkflow(Workspace workspace)
+            : base(workspace)
+        {
+            // Saving previous state so that SonarLint doesn't have to know what's the default state in SonarAnalyzer
+            this.previousShouldExecuteRuleFunc = SonarAnalysisContext.ShouldExecuteRuleFunc;
+            this.previousReportDiagnosticAction = SonarAnalysisContext.ReportDiagnosticAction;
+
+            SonarAnalysisContext.ShouldExecuteRuleFunc = ShouldExecuteVsixAnalyzer;
+            SonarAnalysisContext.ReportDiagnosticAction = VsixAnalyzerReportDiagnostic;
+        }
+
+        internal /* for testing purposes */ bool ShouldExecuteVsixAnalyzer(IAnalysisRunContext context)
+        {
+            if (context.SyntaxTree == null)
+            {
+                return false;
+            }
+
+            Debug.Assert(context.SupportedDiagnostics?.Any() ?? false,
+                "Not expecting a null or empty collection of diagnostic descriptors");
+
+            // Disable the VSIX analyzer as we want the NuGet analyzer to take precedence
+            if (GetProjectNuGetAnalyzerStatus(context.SyntaxTree) != ProjectAnalyzerStatus.NoAnalyzer)
+            {
+                return false;
+            }
+
+            return context.SupportedDiagnostics.Any(d => d.CustomTags.Contains(DiagnosticTagsHelper.SonarWayTag));
+        }
+
+        internal /* for testing purposes */ void VsixAnalyzerReportDiagnostic(IReportingContext context)
+        {
+            Debug.Assert(context.SyntaxTree != null, "Not expecting to be called with a null SyntaxTree");
+            Debug.Assert(context.Diagnostic != null, "Not expecting to be called with a null Diagnostic");
+            Debug.Assert(GetProjectNuGetAnalyzerStatus(context.SyntaxTree) == ProjectAnalyzerStatus.NoAnalyzer,
+                "Not expecting to be called when project contains any SonarAnalyzer NuGet");
+
+            // A DiagnosticAnalyzer can have multiple supported diagnostics and we decided to run the rule as long as at least
+            // one of the diagnostics is enabled. Therefore we need to filter the reported issues.
+            if (context.Diagnostic.Descriptor.CustomTags.Contains(DiagnosticTagsHelper.SonarWayTag))
+            {
+                context.ReportDiagnostic(context.Diagnostic);
+            }
+        }
+
+        #region IDisposable Support
+        private bool disposedValue = false;
+        protected override void Dispose(bool disposing)
+        {
+            if (disposedValue)
+            {
+                return;
+            }
+
+            base.Dispose(disposing);
+            SonarAnalysisContext.ShouldExecuteRuleFunc = this.previousShouldExecuteRuleFunc;
```

Comment:
This code is potentially unsafe; it's relying on the fact that the one workflow class is being disposed before a new one is created. That is what the code is doing at the moment, but ordering dependency isn't obvious so it would be easy to break it by mistake in the future.
It would be safer to check that the static delegate has the expected value before assigning the previous delegate back (either in an _if_, or in an assertion, or both).